pkg/mev/bloxroute_submit_bundle.go

```diff
@@ -0,0 +1,192 @@
+package mev
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type Builder string
+
+const (
+	BuilderBloxroute    Builder = "bloxroute"
+	BuilderFlashbot     Builder = "flashbots"
+	BuilderBeaverBuild  Builder = "beaverbuild"
+	BuilderRsyncBuilder Builder = "rsync-builder"
+	BuilderAll          Builder = "all"
+)
+
+func BloxrouteSubmitBundle( // nolint: cyclop
+	ctx context.Context, c *http.Client, auth, endpoint string,
+	param *BLXRSubmitBundleParams, options ...BloxrouteSubmitBundleOption,
+) (SendBundleResponse, error) {
+	var opts blxrSubmitBundleOptions
+	for _, fn := range options {
+		if fn == nil {
+			continue
+		}
+		fn(&opts)
+	}
+
+	mevBuilders := make(map[Builder]string)
+	if opts.builderBloxroute {
+		mevBuilders[BuilderBloxroute] = ""
+	}
+	if opts.builderFlashbot != nil {
+		sig, err := bloxrouteSignFlashbot(opts.builderFlashbot, param)
+		if err != nil {
+			return SendBundleResponse{}, fmt.Errorf("sign flashbot error: %w", err)
+		}
+		mevBuilders[BuilderFlashbot] = sig
+	}
+	if opts.builderBeaverBuild {
+		mevBuilders[BuilderBeaverBuild] = ""
+	}
+	if opts.builderRsyncBuilder {
+		mevBuilders[BuilderRsyncBuilder] = ""
+	}
+	if opts.builderAll {
```

Comment:
BuilderAll means all builders will be included that includes flashbots builder, do we need to add signature for it?

Reply:
I think we still need to add the flashbot signature separately.

Reply:
Yes, I think we should add the signature for flashbots builder in this case, please update it.

Reply:
I think the caller can use the option flashbot with the option all

Reply:
It's a bit tricky, I think users will not do that because they don't know that or you have to add comment for it.

Reply:
I added a comment for the builderAll option.